File: app/src/main/java/com/anokmik/tripassistant/user/login/LoginFragment.java
package com.anokmik.tripassistant.user.login;

import com.anokmik.tripassistant.R;
import com.anokmik.tripassistant.base.BaseFragment;
import com.anokmik.tripassistant.databinding.FragmentLoginBinding;

public final class LoginFragment extends BaseFragment<FragmentLoginBinding> implements LoginContract.View {

    public static LoginFragment newInstance() {
        return new LoginFragment();
    }

    @Override
    protected boolean displayHomeAsUp() {
        return true;
    }

    @Override
    protected int getTitleResourceId() {
        return R.string.title_login;
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_login;
    }

    @Override
    public void initBinding(FragmentLoginBinding binding) {
        binding.setPresenter(new LoginPresenter(this));
    }

    @Override
    public void back() {
        getActivity().onBackPressed();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/user/login/LoginFragment.kt
package com.anokmik.tripassistant.user.login

import com.anokmik.tripassistant.R
import com.anokmik.tripassistant.base.BaseFragment
import com.anokmik.tripassistant.databinding.FragmentLoginBinding

class LoginFragment : BaseFragment<FragmentLoginBinding>(), LoginContract.View {

    override val displayHomeAsUp = true

    override val layoutId = R.layout.fragment_login

    override val titleResourceId = R.string.title_login

    override fun initBinding(binding: FragmentLoginBinding) {
        binding.presenter = LoginPresenter(this)
    }

    override fun back() = activity.onBackPressed()

    companion object {

        fun newInstance(): LoginFragment {
            return LoginFragment()
        }

    }

}